src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+
+    private T[] arrayList;
+    private int filledWithElements = 0;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialCapacity) {
+        arrayList = (T[]) new Object[initialCapacity];
+    }
+
+    public void resize(int newLength) {
+        T[] newArrayList = (T[]) new Object[newLength];
+        System.arraycopy(arrayList, 0, newArrayList, 0, filledWithElements);
+        arrayList = newArrayList;
+    }
+
     @Override
     public void add(T value) {
-
+        if (filledWithElements == arrayList.length) {
+            resize((int) (arrayList.length * 1.5));
+        }
+        arrayList[filledWithElements++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > filledWithElements) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else if (filledWithElements == arrayList.length) {
```

Comment:
do we need to use else after throw new exception?

Reply:
done